mesonbuild/scripts/env2mfile.py

```diff
@@ -178,13 +240,16 @@ def detect_cross_debianlike(options: T.Any) -> MachineInfo:
         k, v = line.split('=', 1)
         data[k] = v
     host_arch = data['DEB_HOST_GNU_TYPE']
-    host_os = data['DEB_HOST_ARCH_OS']
+    host_os = deb_os_map.get(data['DEB_HOST_ARCH_OS'],
+                             data['DEB_HOST_ARCH_OS'])
```

Comment:
It's a bit tangential, but this pattern of getting seems to be repeated a lot, maybe adding a helper like:
```python
def from_map(mapping: T.Mapping[str, str], name: str) -> str:
   return mapping.get(name, name)

host_os = from_map(deb_os_map, data['DEB_HOST_ARCH_OS'])
```

Reply:
Maybe. My concern about this would be that it isn't necessarily obvious what the default is.

What's actually happening here is something like "identity mapping, with some special cases" so perhaps a slightly longer name than `from_map` could clarify that?

Reply:
Sure, I'm mostly concerned with the code repetition here, and the potential for typos that don't immediately register because we don't use the fallback in our test environment.

Reply:
I called it `replace_special_cases` which I think is clear enough.